fmc/src/hand_off.rs

```diff
@@ -0,0 +1,97 @@
+/*++
+
+Licensed under the Apache-2.0 license.
+
+File Name:
+
+    hand_off.rs
+
+    Implements handoff behavior of FMC :
+        - Retrieves FHT table from fixed address in DCCM.
+        - Transfers control to the runtime firmware.
+++*/
+
+use crate::fmc_env::FmcEnv;
+use caliptra_common::{cprintln, FirmwareHandoffTable};
+use caliptra_drivers::WarmResetEntry4;
+
+#[cfg(feature = "riscv")]
+core::arch::global_asm!(include_str!("transfer_control.S"));
+
+const ICCM_ORG: u32 = 0x40000000;
+const ICCM_SIZE: u32 = 128 << 10;
+
+struct MemoryRegion {
+    start: u32,
+    size: u32,
+}
+
+impl MemoryRegion {
+    fn validate_address(&self, phys_addr: u32) -> bool {
+        phys_addr >= self.start && phys_addr <= self.start + self.size
+    }
+}
+
+const ICCM: MemoryRegion = MemoryRegion {
+    start: ICCM_ORG,
+    size: ICCM_SIZE,
+};
+
+pub struct HandOff {
+    fht: FirmwareHandoffTable,
+}
+
+pub fn dump_fht(fht: &FirmwareHandoffTable) {
+    cprintln!("[fmc] FHT Marker: 0x{:08X}", fht.fht_marker);
+    cprintln!("[fmc] FHT Major Version: 0x{:04X}", fht.fht_major_ver);
+    cprintln!("[fmc] FHT Minor Version: 0x{:04X}", fht.fht_minor_ver);
+    cprintln!("[fmc] FHT Manifest Addr: 0x{:08X}", fht.manifest_load_addr);
+    cprintln!("[fmc] FHT FMC CDI KV KeyID: {}", fht.fmc_cdi_kv_idx);
+    cprintln!(
+        "[fmc] FHT FMC PrivKey KV KeyID: {}",
+        fht.fmc_priv_key_kv_idx
+    );
+    cprintln!(
+        "[fmc] FHT RT Load Address: 0x{:08x}",
+        fht.rt_fw_entry_point_idx
+    );
+    cprintln!(
+        "[fmc] FHT RT Entry Point: 0x{:08x}",
+        fht.rt_fw_load_addr_idx
+    );
+}
+
+impl HandOff {
+    pub fn from_rom() -> Option<HandOff> {
```

Comment:
It looks like RT can also use this function. Maybe `from_previous`? Or if there are any differences, could have a from_rom and from_fmc